qt/scientific_interfaces/Indirect/IndirectFitOutput.cpp

```diff
@@ -0,0 +1,213 @@
+Map mapKeys(const Map &map, const KeyMap &keyMap) {
+  Map newMap;
+
+  for (const auto unmapped : map) {
+    const auto mapping = keyMap.find(unmapped.first);
+    if (mapping != keyMap.end())
+      newMap[mapping->second] = unmapped.second;
+    else
+      newMap[unmapped.first] = unmapped.second;
+  }
+  return newMap;
+}
+} // namespace
+
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace IDA {
+
+IndirectFitOutput::IndirectFitOutput(WorkspaceGroup_sptr resultGroup,
+                                     ITableWorkspace_sptr parameterTable,
+                                     MatrixWorkspace_sptr resultWorkspace,
+                                     const FitDataIterator &fitDataBegin,
+                                     const FitDataIterator &fitDataEnd)
+    : m_resultGroup(resultGroup), m_resultWorkspace(resultWorkspace),
+      m_parameters(), m_outputResultLocations() {
+  addOutput(resultGroup, parameterTable, resultWorkspace, fitDataBegin,
+            fitDataEnd);
+}
+
+IndirectFitOutput::IndirectFitOutput(WorkspaceGroup_sptr resultGroup,
+                                     ITableWorkspace_sptr parameterTable,
+                                     MatrixWorkspace_sptr resultWorkspace,
+                                     IndirectFitData const *fitData,
+                                     std::size_t spectrum) {
+  m_parameters[fitData] = ParameterValues();
+  m_outputResultLocations[fitData] = ResultLocations();
+  addOutput(resultGroup, parameterTable, resultWorkspace, fitData, spectrum);
+}
+
+bool IndirectFitOutput::isSpectrumFit(IndirectFitData const *fitData,
+                                      std::size_t spectrum) const {
+  auto values = m_parameters.find(fitData);
+  return values != m_parameters.end() &&
+         values->second.find(spectrum) != values->second.end();
+}
+
+std::unordered_map<std::string, ParameterValue>
+IndirectFitOutput::getParameters(IndirectFitData const *fitData,
+                                 std::size_t spectrum) const {
+  return findOr(m_parameters, fitData, spectrum,
+                std::unordered_map<std::string, ParameterValue>());
+}
+
+boost::optional<ResultLocation>
+IndirectFitOutput::getResultLocation(IndirectFitData *fitData,
+                                     std::size_t spectrum) const {
+  return findOr(m_outputResultLocations, fitData, spectrum,
+                boost::optional<ResultLocation>(boost::none));
+}
+
+MatrixWorkspace_sptr IndirectFitOutput::getLastResultWorkspace() const {
+  return m_resultWorkspace.lock();
+}
+
+WorkspaceGroup_sptr IndirectFitOutput::getLastResultGroup() const {
+  return m_resultGroup.lock();
+}
+
+void IndirectFitOutput::mapParameterNames(
+    const std::unordered_map<std::string, std::string> &parameterNameChanges,
+    const FitDataIterator &fitDataBegin, const FitDataIterator &fitDataEnd) {
+  for (auto it = fitDataBegin; it < fitDataEnd; ++it) {
+    auto &parameterValues = m_parameters[it->get()];
```

Comment:
Is there any danger here that if the `IndirectFitData` does not already exists in `m_parameters` that a key is added for it, and then for example the `isSpectrumFit` method gives the wrong result..